### DISCUSSION TITLE
YT-Spammers-Purge GUI

### DISCUSSION THREAD

**Discussion by realashleybailey** [ThioJoe/YT-Spammer-Purge#754] [Show and tell]
I wanted to pop by to talk about a new project I am starting to build a more user friendly interactive GUI for YT-Spammer-Purge.
I understand people maybe curious why, so here is some answers, but feel to ask more below.

Why a new GUI?
Even though the current GUI is more than acceptable I believe that the terminal nature of it with the text based interaction may deter would be users who maybe put off by the feel of it.

How am I going solve this?
I am proposing a new Github Repo that will branch off the current work done over here and use the existing infrastructure but with a new GUI layer, though I still need to look more into the current Repos code I am thinking of a Electron App or Web Based App built with a VueJS interface that will pipe commands between the GUI and YT-Spammer-Purge.

What will this do for YT-Spammer-Purge?
Here is some of the things I want to solve with this new GUI:
> Interactive and easy use Settings Menu
> Paginated and interactive comments list
> Oauth integration for easier account linking without API Keys

Please let me know if you want to join the project, it's going to be all Open Source and I hope that I can get Joe's approval and if he sees this I would love to get some insight on the current project.

**Comment by TechStudent10** [comment]
1. Talk about adding a GUI was already discussed in #84 and I'm pretty sure more.
2. While this is a great idea, I'm pretty sure that Joe said that he didn't want to handle user credentials (I think he said that in some early issue about this). So using OAuth is basically out-of-question.
3. That would mean having stuff like auth creds in the repo. This would allow some bad actors to get in and steal user data.

So in conclusion, this is a great idea, just that I guess the security implications outweigh giving users a GUI. Besides, if you were really serious about spam on your channel (especially on a big channel), you probably wouldn't mind the terminal.

**Reply by Firecul** [reply]
> While this is a great idea, I'm pretty sure that Joe said that he didn't want to handle user credentials (I think he said that in some early issue about this). So using OAuth is basically out-of-question.

I think this was more about having an online version of the tool. I could be wrong though.

**Reply by TechStudent10** [reply]
> > While this is a great idea, I'm pretty sure that Joe said that he didn't want to handle user credentials (I think he said that in some early issue about this). So using OAuth is basically out-of-question.
> 
> I think this was more about having an online version of the tool. I could be wrong though.

Yes that is correct. But this is suggesting that a Web-GUI is built. Same thing basically.

**Reply by Firecul** [reply]
I only read Electron App sorry, didn't see web app

**Reply by realashleybailey** [reply]
Very valid points and things I will strongly take into consideration, to detail what you mentioned with handling user credentials, my approach would be using Googles OAuth system to access the users Google Token which would let the program handle comments on behalf of that user, the same approach Joe uses in his tool but instead of people having to setup there own API key I would looking to use my own API Key and pay all the cost of having that API Key.

I think what you mentioned with people being wary of who they give permissions to there Youtube account to is valid and I would feel the same way to and thats why I would want to be very open about the how the software and having it open source, obviously with these kind of things I believe it mainly boils down to individual to individual on what there comfortable with.

I have not seen that discussion yet so I will take a moment to read through and take some notes on what others have discussed.

Thank you to both @TechStudent11 and @Firecul for your feedback.

**Reply by ikkysleepy** [reply]
Sounds great . I don't see any issues with security because you shouldn't have 
auth credentials in the repo and use environment variables. I am working on a similar project that is web/ui driven and focused on sentimental analysis but in order to do that I had to remove spam and unfortunately for me I developed my own spam class before finding this one . This is an interesting project with a lot of good ideas . I think the web interface is what's missing .

**Reply by TechStudent10** [reply]
If this was a web app, instead of building the entire "app" in the window, why not make it a PWA (Progressive Web App(lication)? These are essentially websites, but can be installed to the user's machine if they wish. This would eliminate some possible security risks because of local environment variables on the VPS's side.

**Reply by realashleybailey** [reply]
@TechStudent11 It's great you mentioned PWA because that is actually a big feature that I am going to try and focus on incorporating into the software, if its viable to make it entirely a PWA then it shouldn't be an issue to package the site into an Electron App in the future.

Because I am building the interface in VueJS it has great PWA abilities that make it very easy to work with.

**Reply by realashleybailey** [reply]
@ikkysleepy Yes exactly, I will only ever have access to a users Google Auth Token that can be revoked at any point.
